components/touchscreen/index.rst

```diff
@@ -19,28 +19,82 @@ Base Touchscreen Configuration
     # Example touchscreen
     touchscreen:
       - platform: ...
+        display: display1
         on_touch:
           then:
             ...
+        on_update:
+          then:
+            ...
+        on_release:
+          then:
+            ...
 
 Configuration variables:
 ************************
+- **display** (*Required*, :ref:`config-id`): The display to use this touchscreen with.
 
 - **on_touch** (*Optional*, :ref:`Automation <automation>`): An automation to perform
   when the touchscreen is touched. See :ref:`touchscreen-on_touch`.
-- **display** (**Required**, :ref:`config-id`): The display to use. If only one display is
-  available, this can be omitted.
+- **on_update** (*Optional*, :ref:`Automation <automation>`): An automation to perform
+  when the touchscreen is touched. See :ref:`touchscreen-on_update`.
+- **on_release** (*Optional*, :ref:`Automation <automation>`): An automation to perform
+  when the touchscreen is no longer touched. See :ref:`touchscreen-on_release`.
+
+
+.. _touchscreen-touchpoint:
+
+``TouchPoint`` Argument Type
+----------------------------
+
+Both the :ref:`touchscreen-on_touch` and :ref:`touchscreen-on_update` have an argument of the type :apistruct:`touchscreen::TouchPoint` in a
+list or as variable (only ``on_touch``).
+
+The integer members for the touch positions below are in relation to the display width and height:
+
+- ``x`` and ``y`` are the current position.
+- ``x_last`` and ``y_last`` are the previous position.
+- ``x_first`` and ``y_first`` are the position of the touch when it was first detected.
+- ``x_raw`` and ``y_raw`` are for calibrating the touchscreen in relation of the display. This replaces the properties with the same name in the touchscreen classes.
+
+- ``id`` is a number provided by the touchscreen to uniquely identify the touch on a multi-touch screen.
+- ``state`` indicates the state of the touch. This can be **1**, indicating it is an initial touch, or **2** indicating the touch position has changed/moved.
 
 .. _touchscreen-on_touch:
 
 ``on_touch`` Trigger
 --------------------
 
-This automation will be triggered when the touchscreen detects a touch.
+This automation will be triggered when the touchscreen initially detects a touch on the touchscreen; it will not be fired until
```

Comment:
```suggestion
This automation will be triggered when the touchscreen initially detects a touch on the touchscreen; it will not be fired again until
```